raft/progress.go

```diff
@@ -64,12 +64,17 @@ type Progress struct {
 	RecentActive bool
 
 	// inflights is a sliding window for the inflight messages.
+	// Each inflight message might contain more than one entries
+	// and has a size limit defined in raft config as MaxSizePerMsg.
```

Comment:
Sorry... So a message contains one or multiple raft entries. Each message has a max size.

Say leader sends 8 entries with 1MB per entries. For each message, the max size is 4MB. With flow control, two messages with entries [1,2,3,4] and [5,6,7,8] will be sent. We will have [4, 8] in inflights.

Once follower replies with index 8, then we clear 4 and 8 from inflights together.